### ISSUE TITLE
sql,opt: auto stats collection should include tables in user-defined schemas

### ISSUE THREAD

**Issue by rafiss** [cockroachdb/cockroach#54966]
My reading of the automatic stats code told me that it currently does not update stats for any tables that are in user-defined schemas.

See https://github.com/cockroachdb/cockroach/blob/991893b41e254f2ea5738e4cf4c5cf712c9a4e1e/pkg/sql/stats/automatic_stats.go#L325-L330

This seems like a strong requirement in order for us to release user-defined schemas.

**Comment by RaduBerinde** [comment]
Marking as a release blocker for now. Not a beta blocker.

**Comment by rytaft** [comment]
Looks like we do actually collect automatic stats on these tables, just not upon node startup. Still something that should be fixed, but it's a much smaller issue. Anyway, working on fixing collection on node startup.

**Comment by rytaft** [comment]
I've been playing around with these user-defined schemas, and I think the automatic stats query shown above will still include them even though it has the predicate `schema_name = 'public'`.

```
root@:26257/defaultdb> create schema my_schema;
CREATE SCHEMA

Time: 150ms total (execution 49ms / network 101ms)

root@:26257/defaultdb> create table my_schema.tab(a int primary key);
CREATE TABLE

Time: 120ms total (execution 55ms / network 65ms)

root@:26257/defaultdb> select * from [show tables] where table_name = 'tab';
  schema_name | table_name | type  | estimated_row_count
--------------+------------+-------+----------------------
  my_schema   | tab        | table |                   0
(1 row)

Time: 24ms total (execution 24ms / network 0ms)

root@:26257/defaultdb> select database_name, schema_name, name from crdb_internal.tables where name = 'tab';
  database_name | schema_name | name
----------------+-------------+-------
  defaultdb     | public      | tab
(1 row)

Time: 8ms total (execution 8ms / network 0ms)
```
As shown above, the `schema_name` in `crdb_internal.tables` is still `public`, even though the `schema_name` in `SHOW TABLES` is `my_schema`. It seems that since user-defined schemas are implemented using namespaces, the underlying schema is still `public`.

Am I missing something?

**Comment by rytaft** [comment]
I confirmed with the example schema above that stats for `my_schema.tab` were refreshed upon startup of the node.

I'm going to go ahead and close this, but feel free to re-open if I'm mistaken.

**Comment by rafiss** [comment]
Welp, seems that `crdb_internal.tables` itself is buggy -- those schema_names should not always be public. I filed #54976 

So it seems that bug may have prevented an issue here, but we should probably come back to this and fix this later. (Also, good to know that it would have only affected node startup anyway.)

**Comment by RaduBerinde** [comment]
Do we have tests that verify that tables in user-defined schemas get stats (both on startup and as a result of updates)? If we add those, then any fix to #54976 would necessarily have to fix this issue as well.

**Comment by rytaft** [comment]
No we don't -- I'll add some tests. Re-opening this to track the adding of tests.
